hash_test.go

```diff
@@ -0,0 +1,52 @@
+package bn256
+
+import (
+	"testing"
+
+	"bytes"
+	"math/rand"
+	"strconv"
+)
+
+func TestKnownHashes(t *testing.T) {
+	for i, mh := range marshaledHashes {
+		g := HashG1([]byte{byte(i)})
+		if !bytes.Equal(mh[:], g.Marshal()) {
+			t.Fatal("hash doesn't match a known value")
+		}
+	}
+}
+
+func TestHashG1Collision(t *testing.T) {
+	g := HashG1([]byte(strconv.Itoa(rand.Int())))
+	h := HashG1([]byte(strconv.Itoa(rand.Int())))
+	if *(g.p) == *(h.p) {
+		t.Fatal("found a collision of hashes ")
+	}
+}
+
+func BenchmarkHashG1(b *testing.B) {
+	data := make([][]byte, b.N)
+	for i := 0; i < b.N; i++ {
+		data[i] = []byte(strconv.Itoa(i))
+	}
+
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
```

Comment:
Measuring the time to calculate a hash for _different_ input sizes could be misleading. 
In (https://golang.org/src/crypto/sha256/sha256_test.go +295) shows how to benchmark a hash.

Reply:
Adjusted accordingly.